File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03659/p03659_num3_parsed.py
import sys

def minimize_difference(N, cards):
    cards.sort()
    min_difference = sys.maxsize
    for i in range(1, N):
        snuke_cards = cards[:i]
        raccoon_cards = cards[i:]
        snuke_sum = sum(snuke_cards)
        raccoon_sum = sum(raccoon_cards)
        difference = abs(snuke_sum - raccoon_sum)
        if difference < min_difference:
            min_difference = difference
    return min_difference

N = int(input())
cards = list(map(int, input().split()))

min_difference = minimize_difference(N, cards)
print(min_difference)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03659/p03659_num3_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>

int main() {
    int N;
    std::cin >> N;
    std::vector<long long> cards(N);
    for (int i = 0; i < N; i++) {
        std::cin >> cards[i];
    }

    std::sort(cards.begin(), cards.end());

    long long snuke_sum = 0;
    long long raccoon_sum = 0;

    for (int i = 0; i < N; i++) {
        if (i < N / 2) {
            snuke_sum += cards[i];
        }
        else {
            raccoon_sum += cards[i];
        }
    }

    long long diff = std::abs(snuke_sum - raccoon_sum);

    std::cout << diff;

    return 0;
}